src/main/scala/com/campudus/tableaux/router/BaseRouter.scala

```diff
@@ -41,20 +49,53 @@ trait BaseRouter extends VertxAccess {
 
   val config: TableauxConfig
 
+  val roleModel: RoleModel = RoleModel(config.rolePermissions)
+
   override val vertx: Vertx = config.vertx
 
   /**
     * Base result JSON
     */
   val baseResult: JsonObject = Json.obj("status" -> "ok")
 
-  type AsyncReplyFunction = (=> Future[DomainObject]) => AsyncReply
+  def asyncGetReply(replFunc: => Future[DomainObject])(implicit user: TableauxUser): AsyncReply = {
+    asyncReply(GetReturn)(replFunc)
+  }
 
-  def asyncGetReply: AsyncReplyFunction = asyncReply(GetReturn)(_)
+  def asyncEmptyReply(replFunc: => Future[DomainObject])(implicit user: TableauxUser): AsyncReply = {
+    asyncReply(EmptyReturn)(replFunc)
+  }
 
-  def asyncEmptyReply: AsyncReplyFunction = asyncReply(EmptyReturn)(_)
+  private def enrich(obj: DomainObject, returnType: ReturnType)(implicit user: TableauxUser): JsonObject = {
+    val resultJson = obj.toJson(returnType)
+    obj match {
+      case col: ColumnType[_] =>
+        roleModel.enrichDomainObject(resultJson, ScopeColumn, ComparisonObjects(col.table, col))
+      case colSeq: ColumnSeq => {
+        val seqJson = roleModel.enrichDomainObject(resultJson, ScopeColumnSeq)
+        val columns = colSeq.columns.map(col => enrich(col, returnType))
+        seqJson.mergeIn(Json.obj("columns" -> columns))
+      }
+      case _: ExtendedFolder => roleModel.enrichDomainObject(resultJson, ScopeMedia)
+      case _: Service => roleModel.enrichDomainObject(resultJson, ScopeService)
+      case _: ServiceSeq => roleModel.enrichDomainObject(resultJson, ScopeServiceSeq)
+      case table: Table => roleModel.enrichDomainObject(resultJson, ScopeTable, ComparisonObjects(table))
+      case tableSeq: TableSeq => {
+        val seqJson = roleModel.enrichDomainObject(resultJson, ScopeTableSeq)
+        val tables = tableSeq.tables.map(table => enrich(table, returnType))
+        seqJson.mergeIn(Json.obj("tables" -> tables))
+      }
+      case compTable: CompleteTable => {
+        val tableWithPermissions = enrich(compTable.table, returnType)
+        val columnsWithPermissions = compTable.columns.map(col => enrich(col, returnType))
+        resultJson.mergeIn(tableWithPermissions).mergeIn(Json.obj("columns" -> columnsWithPermissions))
+      }
+      case _ => roleModel.enrichDomainObject(resultJson, null)(user)
```

Comment:
Bin mir nicht sicher ob das nicht auf Fehler laufen würde.
Wäre es nicht besser so wie im alten code in diesem Fall einfach nur das resultJson zurückzugeben?

Reply:
Ja, ist vmtl besser.